platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/UserPropertiesDeletionScheduler.java

```diff
@@ -0,0 +1,164 @@
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserPropertiesDeletionScheduler {
+  static final Path USERS_PROPERTIES_FILE_PATH =
+      Paths.get(new AbsolutePathResolver("etc/users.properties").getPath());
+  private static final Path TEMP_TIMESTAMP_FILE_PATH =
+      Paths.get(new AbsolutePathResolver("data/tmp/timestamp.bin").getPath());
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(UserPropertiesDeletionScheduler.class);
+  private static final String ROUTE_ID = "deletionJob";
+  private final CamelContext context;
+  private boolean usersPropertiesDeletionFlag;
+
+  public UserPropertiesDeletionScheduler(CamelContext camelContext) {
+    this.context = camelContext;
+  }
+
+  public void scheduleDeletion() {
+    if (usersPropertiesDeletionFlag && USERS_PROPERTIES_FILE_PATH.toFile().exists()) {
+      String cron = getCron();
+      if (cron == null) {
+        return;
+      }
+      try {
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .routeId(ROUTE_ID)
+                    .bean(UserPropertiesDeletionScheduler.class, "deleteFiles");
+              }
+            };
+        context.addRoutes(builder);
+        context.start();
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create camel route.", e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  String getCron() {
+    if (!TEMP_TIMESTAMP_FILE_PATH.toFile().exists()) {
+
+      // Create temp file and add timestamp
+      Instant instant = Instant.now();
+      if (!createTempFile(instant)) {
+        return null;
+      }
+      return cronCalculator(instant);
+
+    } else {
+
+      // Read timestamp
+      Instant firstInstall = installationDate();
+      if (firstInstall == null) {
+        return null;
+      }
+
+      if (Instant.now().isAfter(firstInstall.plus(Duration.ofDays(3)))) { // Timestamp expired
+        deleteFiles();
+        return null;
+      }
+
+      return cronCalculator(firstInstall);
+    }
+  }
+
+  private boolean createTempFile(Instant instant) {
+    try {
+      if (TEMP_TIMESTAMP_FILE_PATH.toFile().createNewFile()) {
+        return writeTimestamp(instant);
+      }
+    } catch (IOException e) {
+      LOGGER.debug("Unable to access the temporary file", e);
+    }
+    return false;
+  }
+
+  private boolean writeTimestamp(Instant instant) throws IOException {
+    try (ObjectOutputStream objectOutputStream =
+        new ObjectOutputStream(new FileOutputStream(TEMP_TIMESTAMP_FILE_PATH.toFile()))) {
+      objectOutputStream.writeObject(instant);
```

Comment:
Is there a reason we are writing out an Instance java object instead of just writing out a long representing the milliseconds? I generally try to avoid writing objects to out.

Reply:
I'd suggested the object output instead of marshalling and unmarshalling longs, parsing them as strings, reinterpreting them, etc.